### PR TITLE
Refactor doctrine.event_subscriber to doctrine.event_listener for Symfony 7 compatibility

### PR DESCRIPTION
**Pull request by TheCadien**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | yes
| Deprecations? | no 
| Fixed tickets | fixes [#7156](https://github.com/sulu/sulu/pull/7156#issuecomment-2060676755)
| License | MIT
| Documentation PR | sulu/sulu-docs#

### CODE REVIEW

**Comment by alexander-schranz** [comment]
@TheCadien Please rebase to contain only the commits changing the event subscribers to event listeners.

File: src/Sulu/Bundle/ActivityBundle/Infrastructure/Doctrine/Listener/DomainEventCollectorListener.php [hunk 1]
```diff
@@ -9,15 +9,13 @@
...
+namespace Sulu\Bundle\ActivityBundle\Infrastructure\Doctrine\Listener;
 
-use Doctrine\Common\EventSubscriber;
 use Doctrine\ORM\Event\OnClearEventArgs;
 use Doctrine\ORM\Event\PostFlushEventArgs;
-use Doctrine\ORM\Events;
 use Sulu\Bundle\ActivityBundle\Application\Collector\DomainEventCollectorInterface;
 
-class DomainEventCollectorSubscriber implements EventSubscriber
+class DomainEventCollectorListener
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/ActivityBundle/Infrastructure/Doctrine/Listener/DomainEventCollectorListener.php]: for BC reasons I would currently keep the class names

File: src/Sulu/Bundle/PersistenceBundle/Resources/config/services.xml [hunk 2]
```diff
@@ -11,20 +11,20 @@
...
-            <tag name="doctrine.event_subscriber"/>
+            <tag name="doctrine.event_listener"/>
         </service>
 
         <service id="sulu.persistence.event_subscriber.orm.user_blame" class="%sulu.persistence.event_subscriber.orm.user_blame.class%">
             <argument type="service" id="security.token_storage" on-invalid="null"/>
 
             <!-- Priority 50 as the MetadataLoader need to be before the ResolveTargetEntityListener of Doctrine -->
-            <tag name="doctrine.event_subscriber" priority="50"/>
+            <tag name="doctrine.event_listener" priority="50"/>
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/PersistenceBundle/Resources/config/services.xml]: requires `event`

File: src/Sulu/Bundle/PersistenceBundle/Resources/config/services.xml [hunk 3]
```diff
@@ -11,20 +11,20 @@
...
         </service>
 
         <service id="sulu.persistence.event_subscriber.orm.metadata" class="%sulu.persistence.event_subscriber.orm.metadata.class%">
             <argument>%sulu.persistence.objects%</argument>
             <!-- Priority 8000 as the MetadataLoader need to be before all other Doctrine listeners -->
-            <tag name="doctrine.event_subscriber" priority="8000" />
+            <tag name="doctrine.event_listener" priority="8000" />
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/PersistenceBundle/Resources/config/services.xml]: requires `event`

**Review by alexander-schranz — CHANGES_REQUESTED** [review]
For bc reasons I would keep all class names at current state.

And some of the tags are still missing the `event` definitions.

**Comment by TheCadien** [comment]
@alexander-schranz done

File: src/Sulu/Bundle/SecurityBundle/Resources/config/services.xml [hunk 4]
```diff
@@ -342,10 +342,10 @@
 
         <service
             id="sulu_security.permission_inheritance_subscriber"
-            class="Sulu\Bundle\SecurityBundle\EventListener\PermissionInheritanceSubscriber"
+            class="PermissionInheritanceSubscriber"
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/SecurityBundle/Resources/config/services.xml]: unexpected change

File: src/Sulu/Bundle/SecurityBundle/Resources/config/2fa_force.xml [hunk 5]
```diff
@@ -3,10 +3,11 @@
            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
            xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
     <services>
-        <service id="sulu_security.force_two_factor_listener" class="Sulu\Bundle\SecurityBundle\EventListener\ForceTwoFactorSubscriber">
+        <service id="sulu_security.force_two_factor_listener" class="ForceTwoFactorSubscriber">
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/SecurityBundle/Resources/config/2fa_force.xml]: unexpected change

**Review by alexander-schranz — CHANGES_REQUESTED** [review]

**Review by alexander-schranz — APPROVED** [review]

**Comment by alexander-schranz** [comment]
@TheCadien Thank you!

**Pull request merged**